src/Documents/Import/PageImporter.php

```diff
@@ -51,4 +69,26 @@ public function readYamlFileAndSetParameters(string $filename, array $params = [
 
         return '';
     }
+
+    /**
+     * @param array<string, array<string, mixed>> $configPages
+     */
+    private function checkAndUpdatePage(Page $page, array &$configPages): void
+    {
+        $oldPath = $page->getPath();
+        $existingParent = Document::getById($page->getParentId() ?? -1);
+        if (!$existingParent) {
+            $existingParent = Document::getByPath($page->getPath() ?? '');
+            if (!$existingParent) {
+                throw new \InvalidArgumentException('Neither parentId nor path leads to a valid parent element');
+            }
+            $page->setParentId($existingParent->getId());
+            $newPath = $existingParent->getPath() . $page->getKey() . '/';
+            foreach ($configPages as $configPage) {
+                if (\array_key_exists('path', $configPage[YamlExportPage::PAGE]) && \is_string($oldPath)) {
+                    str_replace($oldPath, $newPath, $configPage[YamlExportPage::PAGE]['path']);
```

Comment:
Neben dem refactoring das ich unten vorgschlagen habe, muss zuätzlich dieser Teil hier noch gefixt werden.
Das pass by reference klappt so nicht. Siehe hier https://onlinephp.io/c/ce4a2

Ich habe überlegt einen Test dafür zu schreiben, aber die `parseYaml` methode tut zu viel. Kannst du vielleicht das yaml parsen und den Aufruf für `checkAndUpdatePage` kapseln? Vielleicht in zwei Klassen? Dann wäre es wesentlich einfacher testbar.

Reply:
That's true.
I didn't test the import process obviously - so I'll improve that.

But the suggestions for the refactoring of method names and separations of concerns will create a breaking change?!

If we are fine with that, I'll can do.

Reply:
For the first glance, i would prefer to not break backwards compatibility.

What about trigger deprecation using https://github.com/symfony/deprecation-contracts
```
function parseYaml() {
    trigger_deprecation('teamneusta/pimcore-import-export-bundle', '1.2', 'Using "%s" is deprecated, use "%s" instead.', 'parseYaml', 'importYaml');
    ...
}
```
plus
```
function importYaml() {
    $this->parseYaml();
}
```
Something like this. If you are willing to wait until Thuesday, let us discuss it then